.github/workflows/scripts/scanner-fetch-nvd.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+base_url="https://services.nvd.nist.gov/rest/json/cves/2.0"
+results_per_page=2000
+dir_name="nvd-data"
+
+# Function to fetch data for a given period
+fetch_data_for_period () {
+  local period_start_date=$1
+  local period_end_date=$2
+  local current_start_index=0
+
+  while : ; do
+    local file_name="${period_start_date}_${current_start_index}.json"
+
+    echo "Fetching data from $period_start_date to $period_end_date, starting at index $current_start_index..."
+
+    retry_intervals=(6 12 24) # Array of retry intervals in seconds
+    success=false
+
+    for i in "${retry_intervals[@]}"; do
+        if curl -s --fail --show-error "$base_url?pubStartDate=$period_start_date&pubEndDate=$period_end_date&startIndex=$current_start_index" -o "$dir_name/$file_name"; then
+            success=true
+            break # Exit
+        else
+            sleep "$i" # Wait for next retry
+        fi
+    done
+
+    if [ "$success" = false ]; then
+        echo "Failed after retries"
+        exit 1
+    fi
+
+    if [ ! -s "$dir_name/$file_name" ] || ! jq empty "$dir_name/$file_name"; then
+      echo "Downloaded file is empty or invalid for the period: $period_start_date to $period_end_date, starting index $current_start_index"
+      exit 1
+    fi
+
+    local total_results
+    total_results=$(jq '.totalResults' "$dir_name/$file_name") || { echo "jq command failed"; exit 1; }
+
+    if ((current_start_index >= total_results)); then
+      break
+    fi
+
+    current_start_index=$((current_start_index + results_per_page))
+    sleep 6
+  done
+}
+
+if [ ! -d "$dir_name" ]; then
+  mkdir "$dir_name"
+else
+  echo "Directory $dir_name exists."
+fi
+
+for (( year=2008; year<=$(date +%Y); year++ )); do
+    START_DATE="${year}-01-01T00:00:00.000"
+    END_YEAR=$((year + 1))
+
+    while [ "$START_DATE" \< "${END_YEAR}-01-01T00:00:00.000" ]; do
+        END_DATE=$(date +%Y-%m-%d -d "${START_DATE%T*} + 119 days")T00:00:00.000
```

Comment:
NVD Api has constraint to fetch data at maximum 120 days.

Reply:
I think it's worth adding a comment about that in the script.